src/hardware/serialport/serialport.cpp

```diff
@@ -1229,16 +1229,16 @@ bool CSerial::Putchar(Bit8u data, bool wait_dsr, bool wait_cts, Bitu timeout) {
 class SERIALPORTS:public Module_base {
 public:
 	SERIALPORTS (Section * configuration):Module_base (configuration) {
-		Bit16u biosParameter[4] = { 0, 0, 0, 0 };
+		Bit16u biosParameter[SERIAL_MAX_PORTS] = {0};
 		Section_prop *section = static_cast <Section_prop*>(configuration);
 
 #if C_MODEM
 		const Prop_path *pbFilename = section->Get_path("phonebookfile");
 		MODEM_ReadPhonebook(pbFilename->realpath);
 #endif
 
-		char s_property[] = "serialx"; 
-		for(Bitu i = 0; i < 4; i++) {
+		char s_property[] = "serialx";
+		for (uint8_t i = 0; i < SERIAL_MAX_PORTS; ++i) {
 			// get the configuration property
 			s_property[6] = '1' + i;
```

Comment:
Interesting… So if someone in the future defines SERIAL_MAX_PORTS to 9 or more, then `s_property` will have values … `serial9`, `serial:`, `serial;`, `serial<`, … ;)

Reply:
It would worked up to a point, and then break in the serial code that looks at port addresses and irq, which are hard-coded to limits of 4. Like this:
case 0x3f8: i=0; break;
		case 0x2f8: i=1; break;
		case 0x3e8: i=2; break;
		case 0x2e8: i=3; break;

To really make it work, we would have to programmaticly extended this logic to N ports.

So the goal of this define is to remove a magic number in the code.  The intent is still a hard-coded cap of 4 ports.  

I will add a small comment to this effect at the define (just so users don't think they can crank it up and it will work).

Reply:
Added a note in fixup commit 731c2999.